src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -89,9 +89,11 @@ async function syncClient(
     // If an existing client is found, update it
     if (token && !isRetry) {
       Log.debug(pkg.metadata, `Found existing token for ${clientReq.clientId}`);
+      handleClientGroups(clientReq);
       client = await apiCall(clientReq, "PUT", token);
     } else {
       Log.debug(pkg.metadata, `Creating new client for ${clientReq.clientId}`);
+      handleClientGroups(clientReq);
       client = await apiCall(clientReq);
     }
```

Comment:
This is in both sides of the if/else here. Pull out `handleClientGroups()` and put it above the if/else.

Reply:
updated